script/hassfest/triggers.py

```diff
@@ -0,0 +1,232 @@
+"""Validate triggers."""
+
+from __future__ import annotations
+
+import contextlib
+import json
+import pathlib
+import re
+from typing import Any
+
+import voluptuous as vol
+from voluptuous.humanize import humanize_error
+
+from homeassistant.const import CONF_SELECTOR
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv, selector, trigger
+from homeassistant.util.yaml import load_yaml_dict
+
+from .model import Config, Integration
+
+
+def exists(value: Any) -> Any:
+    """Check if value exists."""
+    if value is None:
+        raise vol.Invalid("Value cannot be None")
+    return value
+
+
+FIELD_SCHEMA = vol.Schema(
+    {
+        vol.Optional("example"): exists,
+        vol.Optional("default"): exists,
+        vol.Optional("required"): bool,
+        vol.Optional(CONF_SELECTOR): selector.validate_selector,
+    }
+)
+
+TRIGGER_SCHEMA = vol.Any(
+    vol.Schema(
+        {
+            vol.Optional("fields"): vol.Schema({str: FIELD_SCHEMA}),
+        }
+    ),
+    None,
+)
+
+TRIGGERS_SCHEMA = vol.Schema(
+    {
+        vol.Remove(vol.All(str, trigger.starts_with_dot)): object,
+        cv.slug: TRIGGER_SCHEMA,
+    }
+)
+
+NON_MIGRATED_INTEGRATIONS = {
+    "calendar",
+    "conversation",
+    "device_automation",
+    "geo_location",
+    "homeassistant",
+    "knx",
+    "lg_netcast",
+    "litejet",
+    "persistent_notification",
+    "samsungtv",
+    "sun",
+    "tag",
+    "template",
+    "webhook",
+    "webostv",
+    "zone",
+    "zwave_js",
```

Comment:
Some of these have custom UI, should they be excluded from `triggers_platforms/subscribe`?

Reply:
why exclude them? the frontend already knows which are the ones that have custom ui, so it can filter them out if needed, right? 
and this way we offer a consistent way to get all of them.

Reply:
OK